### DISCUSSION TITLE
How to limit the number of parallel containers

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [testcontainers/testcontainers-dotnet#1017] [Q&A] [answered]
Hi all, one question:
I have hundreds of tests, in ~30 test collections, I`m using one container per test collection/test class, you already got the point, 30 containers running at the same time...

Is there a way to limit the number of parallel containers?

Thank you very much!

**Comment by HofmeisterAn** [comment, marked as answer]
This depends on the test framework you are using, they use different ways to configure it (there is nothing built-in in Testcontainers). If you need the same service again and again, it is recommended to create the container once (singleton) and share it across the tests. This will make your tests run faster and use fewer resources.
